TextEditorExtrasSettingsForm.inc.php

```diff
@@ -24,9 +24,23 @@ public function __construct($plugin) {
 	 * Settings are stored by context, so that each journal or press
 	 * can have different settings.
 	 */
-	public function initData() {
+	public function initData()
+	{
 		$contextId = Application::get()->getRequest()->getContext()->getId();
-		$this->setData('additions', $this->plugin->getSetting($contextId, 'additions'));
+		$additions = [
+			"masthead" => ["description" => []],
+			"authorGuidelines" => ["authorGuidelines" => [], ["copyrightNotice" => []]],
+			"license" => ["licenseTerms" => []],
+			"reviewerGuidance" => ["reviewGuidelines" => [], "competingInterests" => []],
+			"editEmailTemplate" => ["body" => []],
+			"titleAbstract" => ["abstract" => []],
+			"announcement" => ["description" => [], "descriptionShort" => []]
+		];
+		$settings = $this->plugin->getSetting($contextId, 'additions');
+		if($settings){
+			$additions = array_merge_recursive($additions, $settings);
```

Comment:
`$this->plugin->getSetting()` might not return an array, so can you check that it's an array before passing it to `array_merge_recursive()`?

```php
if (is_array($settings)) {
```

Thanks!